code/modules/hydroponics/gene_modder.dm

```diff
@@ -512,10 +519,11 @@
 	if(!HAS_TRAIT(src, TRAIT_CMAGGED))
 		to_chat(user, "<span class='warning'>The bananium ooze flips a couple bits on the plant disk's display, making it look just like the..!</span>")
 		ADD_TRAIT(src, TRAIT_CMAGGED, "clown_emag")
-		update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON)
+		update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON|UPDATE_OVERLAYS)
+		playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
 
 /obj/item/disk/plantgene/uncmag()
-	update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON)
+	update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON|UPDATE_OVERLAYS)
```

Comment:
`UPDATE_OVERLAYS` is included as part of `UPDATE_ICON` here, I'm also pretty sure update_appearance() only supports these 3 flags `UPDATE_NAME|UPDATE_DESC|UPDATE_ICON`. Alternatively, update_appearance() with no flags applies the `ALL` flag (which toggles all flags to true), which you could do instead, but maybe the way you have it with the flags shown is more readable.

Reply:
`update_appearance()` supports the whole range of bitflags. `UPDATE_ICON` is simply both `UPDATE_ICON_STATE` and `UPDATE_OVERLAYS` : `#define UPDATE_ICON (UPDATE_ICON_STATE|UPDATE_OVERLAYS)`

Reply:
I just meant here. Update appearances doesnt do anything with the overlays itself. `UPDATE_ICON` has to be called specifically from what I understand
https://github.com/ParadiseSS13/Paradise/blob/cedc12f64c67ea50b1b537eca91e7e13f5a3a58d/code/game/atoms.dm#L397-L408

Reply:
`UPDATE_ICON` is defined as both, and just passing one will be truthy for `if(updates & UPDATE_ICON)` and from there will be the equivalent of `update_icon(UPDATE_ICON_STATE)` or `update_icon(UPDATE_OVERLAYS)` as `updates` is what gets passed around.